### PR TITLE
Windows test framework fixes

### PR DESCRIPTION
**Pull request by sschuberth**
First round of fixes to the test framework to make tests basically run on Windows. With this, the remaining failing tests are:

In *test/test-custom-transfers.sh*:

    test: custom-transfer-upload-download ...                          FAILED

In *test/test-fetch-include.sh*:

    test: fetch: include first matching file ...                       FAILED
    test: fetch: include second matching file ...                      FAILED

Running tests in AppVeyor will be enabled in a separate PR once these tests are fixed and https://github.com/appveyor/ci/issues/1022 is deployed.

### CODE REVIEW

File: script/integration.go [hunk 1]
```diff
@@ -196,7 +196,7 @@ func setBash() {
 		os.Exit(1)
 	}
 
-	bashPath = strings.TrimSpace(string(out))
+	bashPath = strings.TrimSpace(strings.Split(string(out), "\n")[0])
```
Comment (technoweenie) [inline script/integration.go]: Calling `strings.Split()` on an empty string returns an empty `[]string`, so accessing `[0]` will cause go to panic. That's probably a super rare scenario, but I think it's worth adding a check right above this line:

```go
if len(out) == 0 {
    fmt.Printf("No output from '%s bash'\n", findcmd)
    os.Exit(1)
}

**Comment by technoweenie** [comment]
Hey @sschuberth, thanks so much for this. As you can probably tell, most of us don't use Windows much, so this helps a lot. Apparently I don't use bash often either, not sure why I didn't export `GIT_CONFIG_NOSYSTEM` :trollface: 

I can merge this once you take care of the small change I mentioned. 

I'm traveling this week, so I can't look at the 3 remaining windows failures. I'll take a look when I get home to my windows machine.

**Comment by sschuberth** [comment]
@technoweenie Done. I was planning to look at the remaining Windows failures myself, but I still wanted to get these fixes in already, just in case others want to have a look at the failures, too.

**Comment by sschuberth** [comment]
I've pushed some more commits on top, which among others fix `test: custom-transfer-upload-download`. However, on my Windows 10 machine I'm now seeing more failing tests than before:

    test: env with no remote ...                                       FAILED
    test: env with origin remote ...                                   FAILED
    test: env with multiple remotes ...                                FAILED
    test: env with other remote ...                                    FAILED
    test: env with multiple remotes and lfs.url config ...             FAILED
    test: env with multiple remotes and lfs configs ...                FAILED
    test: env with multiple remotes and lfs url and batch configs ...  FAILED
    test: env with .lfsconfig ...                                      FAILED
    test: env with .gitconfig ...                                      FAILED
    test: env with environment variables ...                           FAILED
    test: env with bare repo ...                                       FAILED
    test: env with skip download errors ...                            FAILED
    test: env with extra transfer methods ...                          FAILED
    test: fetch: include first matching file ...                       FAILED
    test: fetch: include second matching file ...                      FAILED
    test: fsck default ...                                             FAILED
    test: submodule env ...                                            FAILED
    test: track ...                                                    FAILED
    test: track representation ...                                     FAILED
    test: git worktree ...                                             FAILED

I'll try to look into these one by one ...

**Comment by sschuberth** [comment]
Added more fixes, now failures on my Windows 10 machine are down to:

    test: fetch: include first matching file ...                       FAILED
    test: fetch: include second matching file ...                      FAILED
    test: track ...                                                    FAILED
    test: track representation ...                                     FAILED

**Comment by sschuberth** [comment]
@ttaylorr @technoweenie Even if this does not fix all tests on Windows yet, can we get the current state merged? I have more local commits in the works, but I'd prefer to get in what I have so far to clean things up a bit.

**Comment by ttaylorr** [comment]
Sure, I don't see a problem with merging this in an incremental state since we're not predicating our merges on Windows tests in any way. I want to make sure this is cool with @technoweenie too, so I'll defer to his judgement on this one. 😄

**Comment by sschuberth** [comment]
@ttaylorr I guess @technoweenie's thumbs-up to your post means we are good to merge? If so, please have another look at the diff to double check, and hit the merge button if all is good! :-)

**Pull request merged**